Src/Web/Web.Shared.Net/RequestTrackingTelemetryModule.cs

```diff
@@ -211,6 +212,12 @@ public void OnEndRequest(HttpContext context)
 
             if (this.childRequestTrackingSuppressionModule?.OnEndRequest_ShouldLog(context) ?? true)
             {
+                var intermediateRequest = (RequestTelemetry)context.Items[IntermediateRequestHttpContextKey];
+                if (intermediateRequest != null)
```

Comment:
Is there a guarantee that internal request will reach this OnEndRequest handler faster all the time?
I remember that with TransferRequestHandler we didn't have such a guarantee as per @MS-TimothyMothra.
Without that, the tests may still fail very rarely but customers with high load of requests would sometimes see both reported vs. one reported.